test-utils/cli/src/lib.rs

```diff
@@ -35,6 +35,35 @@ use std::{
 };
 use tokio::io::{AsyncBufReadExt, AsyncRead};
 
+/// Similar to [`crate::start_node`] spawns a dev node on port `45789`, but works in environments
+/// where the substarte binary is not accessible with `cargo_bin("substrate-node")`.
```

Comment:
It might be worth having a look at the [substrate-runner](https://github.com/paritytech/subxt/blob/9cfac4eec7571094f8f6ea864f6e662cf9737304/testing/substrate-runner/src/lib.rs) crate from subxt, a zero dependency crate, it has a lovely builder pattern to spawn a substrate-binary from paths with custom arguments. Cargo-contract also uses the crate for testing purposes.

It can retrieve the ws-port (p2p id, address, and port), so there would be no need to hardcode it to `45789` such multiple instances can run in parallel.

Reply:
Thank you for mentioning this, the tool looks awesome!

It still seems limited in that it requires a substrate binary is present on the path, which I'm trying to avoid. I wonder if it could be improved by calling into the `node-template` lib like I am allowing here, removing that requirement?

I'll make the command more flexible to customise the args in case we want to experiment with some use cases like this in the future.